EuclideanGeometry/Foundation/Construction/Triangle/IsocelesTriangle.lean

```diff
@@ -12,15 +25,56 @@ variable {P : Type _} [EuclideanPlane P]
 
 section IsocelesTriangles
 
+open Triangle
 def Triangle.IsIsoceles (tri : Triangle P) : Prop := tri.edge₂.length = tri.edge₃.length
 
-theorem is_isoceles_tri_iff_ang_eq_ang_of_nd_tri {tri_nd : Triangle_nd P} : tri_nd.1.IsIsoceles ↔ (tri_nd.angle₂.value= - tri_nd.angle₃.value) := sorry
+
+
+
+theorem is_isoceles_tri_iff_ang_eq_ang_of_nd_tri {tri_nd : Triangle_nd P} : tri_nd.1.IsIsoceles ↔ (tri_nd.angle₂.value=  tri_nd.angle₃.value) := by
 -- To show angle equal => sides equal, use anti-congruent of the triangle with itself. SAS
+rcases Triangle.edge_eq_edge_of_flip_vertices tri_nd.1 with ⟨a, b, c⟩
+rcases Triangle.angle_eq_neg_angle_of_flip_vertices tri_nd with ⟨x, y, z⟩ 
+have h21 :  tri_nd.1.edge₃.length = tri_nd.flip_vertices.1.edge₂.length := c
+have h22 : tri_nd.1.edge₂.length = tri_nd.flip_vertices.1.edge₃.length := b
+have h23 : tri_nd.angle₁.value = - tri_nd.flip_vertices.angle₁.value := x
+have h24 : tri_nd.angle₃.value = - tri_nd.flip_vertices.angle₂.value := z
+
+constructor
+intro h0
+have h1 : tri_nd.1.edge₂.length = tri_nd.1.edge₃.length := by
+  exact h0  
+have _ : tri_nd.1 IsACongrTo tri_nd.flip_vertices.1 := by
+  apply acongr_of_SAS
+  rw[← h21 , h1]
+  rw[h23]
+  rw[← h22 , ← h1]
+have h4 : tri_nd.angle₂.value = - tri_nd.flip_vertices.angle₂.value := by
+  rw[IsACongr.angle₂.value]
+rw[h4 , h24]
+
+intro k0
+have _ : tri_nd.1 IsACongrTo tri_nd.flip_vertices.1 := by
+  apply acongr_of_ASA
+  rw[k0 , h24]
+  exact a
+  rw[← k0 , y]
+have k2 : tri_nd.1.edge₂.length = tri_nd.flip_vertices.1.edge₂.length := by
+  rw[IsACongr.edge₂]
+rw[← h21] at k2
+exact k2
+
 
 namespace Triangle
 
 -- Changing the order of vertices 2 and 3 in an isoceles triangle does not change the property of being an isoceles triangle.
-theorem is_isoceles_of_flip_vert_isoceles (tri : Triangle P) (h : tri.IsIsoceles) : tri.flip_vertices.IsIsoceles := by sorry
+theorem is_isoceles_of_flip_vert_isoceles (tri : Triangle P) (h : tri.IsIsoceles) : tri.flip_vertices.IsIsoceles := by 
+have h₁ : tri.edge₂.length = tri.edge₃.length := by 
+  exact h
+have h₂ : tri.flip_vertices.edge₂.length = tri.flip_vertices.edge₃.length := by
+  rcases Triangle.edge_eq_edge_of_flip_vertices tri with ⟨_, b, c⟩ 
+  rw[← b , ← c , h₁]
```

Comment:
do here needs `h\1`? Is `h` OK? If not, you can write `unfold h`, instead of `have`